src/modules/Packages.rb

```diff
@@ -2072,19 +2079,19 @@ def SelectProduct
 
       products = Pkg.ResolvableProperties("", :product, "")
 
-      if products.empty?
+      if !products || products.empty?
         log.info("No product found on media")
         return true
       end
 
       # no product selected -> select them all
       ret = true
-      unless products.find_index { |p| p["status"] == :selected }
+      unless products.any? { |p| p["status"] == :selected }
         log.info("No product selected so far...")
         selected_products = []
         products.each do |p|
           product_name = p["name"] || ""
-          if Builtins.regexpmatch(product_name, "-migration$")
+          if product_name.match(/-migration$/)
```

Comment:
Just a note: in Ruby the end of string matcher is `\z`, `$` means end of line (`\n`). But as the product name cannot be multi-line it does not matter in this case...

(The Ruby way is a bit complicated, see http://ruby-doc.com/docs/ProgrammingRuby/html/language.html#UL for details.)